apis/python/src/tiledbsoma/soma_vfs.cc

```diff
@@ -104,10 +104,22 @@ void load_soma_vfs(py::module& m) {
             "open",
             [](SOMAVFSFilebuf& buf, const std::string& uri) {
                 return buf.open(uri, std::ios::in);
-            })
-        .def("read", &SOMAVFSFilebuf::read, "size"_a = -1)
-        .def("tell", &SOMAVFSFilebuf::tell)
-        .def("seek", &SOMAVFSFilebuf::seek, "offset"_a, "whence"_a = 0)
+            },
+            py::call_guard<py::gil_scoped_release>())
+        .def(
+            "read",
+            &SOMAVFSFilebuf::read,
+            "size"_a = -1,
+            py::call_guard<py::gil_scoped_release>())
```

Comment:
The code for read creates a python bytes array, which requires the GIL.  So you will need to release/acquire the GIL within the read() method, not outside of it.